src/Symfony/Component/Utf8/CodePoints.php

```diff
@@ -0,0 +1,234 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Utf8;
+
+use Symfony\Component\Utf8\Exception\InvalidArgumentException;
+
+final class CodePoints implements GenericStringInterface, \Countable
+{
+    use Utf8Trait;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function count(): int
+    {
+        return mb_strlen($this->string, 'UTF-8');
```

Comment:
Oh well, the missing `private` for `const` in PHP 7. A `private static` would do as well. I mean, it violates the DRY principle (no offense, but many constructs do in this implementation), but other than that I mainly see a potential source for bugs (typos).